### DISCUSSION TITLE
Metamask doesn't open when using auth core on mobile browser

### DISCUSSION THREAD

**Discussion by whpoole** [Web3Auth/Web3Auth#776] [Help]
Hi everyone,

Currently having an issue with using @web3auth/core and Metamask. It works fine on desktop browsers, but it fails to redirect to the Metamask app when attempting to authenticate on a mobile browser (eg IOS Safari). 

I've got a pretty basic setup with a metamask adapter configured:

`const metamaskAdapter = new MetamaskAdapter({})`

`web3auth.configureAdapter(metamaskAdapter)`

and then connecting with `auth.connectTo('metamask')`

As I say, this works great on desktop browsers. On mobile, i'd expect metamask to open, ask to connect to the site, and then redirect to the site. However, nothing happens when running 'connectTo'. Social logins work fine, just not Metamask.

Looking at console logs from the mobile browser, it looks like it's not detecting any injected window.ethereum and, as a result, the adapter is not becoming 'ready' to run. Is there a way to get the expected redirect to metamask -> connect -> return to site functionality working?

Thanks!

<img width="501" alt="Screenshot 2022-10-26 at 11 11 13" src="https://user-images.githubusercontent.com/66960829/198000341-eef9b12d-885e-466c-98f3-f9bb236d5136.png">

<img width="502" alt="Screenshot 2022-10-26 at 11 11 21" src="https://user-images.githubusercontent.com/66960829/198000359-5d4ea77c-d9f5-4eee-8745-cd430f0cf32b.png">

**Comment by BboyStatix** [comment]
Hey there, that error occurs due to there not being any metamask extension installed on the browser. And if I'm not mistaken there is currently no metamask extension for Safari https://metamask.io/download/

EDIT: Answer in comment below

**Reply by whpoole** [reply]
Hi there,

You’re right that there isn’t a safari extension, however for most integrations when using safari on IOS you get redirected to the native MetaMask app to connect and sign transactions. Seems this works fine using the standard web3auth modal, judging by the example app, Just wondering if this is currently possible with web3auth core. 

Thanks!

**Reply by BboyStatix** [reply]
@whpoole I've figured out the problem. Basically it works for the example app because it is utilising metamask via WalletConnect.
As you can see in the pic below, the metamask icon is inside the WalletConnect section
![198493225-a8e88130-5d6b-42d8-94aa-5cf849bea19b](https://user-images.githubusercontent.com/19769879/198494233-f69fccc1-fea4-49f1-8702-932758ca35ca.jpeg)

The metamask adapter on the other hand only works with the metamask extension installed. So on a browser without the metamask extension, you'd get the aforementioned error and the metamask icon wouldn't display (except for the one displayed via WalletConnect). You can try this out yourself by opening the [demo app](https://demo-app.web3auth.io/) in an incognito  window.

That's why in the demo app code, web3auth is being inited like so https://github.com/Web3Auth/web3auth-web/blob/da7b8f8784672c8e33f35f9314bed5127d1b7131/demo/vue-app/src/chains/ethereum.vue#L159-L166
As you can see `showOnMobile` is set to `false` because it doesn't work on mobile browsers. You don't actually have to pass in these options yourself actually because the default values for metamask and phantom are already set to `false` for `showOnMobile`. This was just to give you an idea.
